src/sonolus/playlist/info.ts

```diff
@@ -1,23 +1,25 @@
 import { Icon, Text } from '@sonolus/core'
 import { randomize } from '../../utils/math.js'
 import { sonolus } from '../index.js'
+import { hideSpoilers } from '../utils/spoiler.js'
 import { playlistSearches } from './search.js'
 
 export const installPlaylistInfo = () => {
-    sonolus.playlist.infoHandler = () => {
+    sonolus.playlist.infoHandler = ({ options }) => {
+        const filteredPlaylists = hideSpoilers(options.spoilers, sonolus.playlist.items)
```

Comment:
Levels in playlists also need to be filtered here. Playlist level filtering code can be extracted and reused in related endpoints.